Choose network for hardware wallets

jessgusclark opened the pull request:
Allows the user to choose the network when selecting hardware wallets.

![Screen Shot 2021-09-13 at 4 08 41 PM](https://user-images.githubusercontent.com/766679/133089250-c3c97465-a8a1-4ceb-969c-323caebefb2f.png)

See detailed commit messages for full details.

## provider controller

To accomplish this the provider controller has been taken from web3modal (src/controllers/providers.ts). The majority of this file has stayed the same, the following things were added or changed:

* `onClick` method accepts an optional parameter with RPCURL and ChainId [here](https://github.com/rsksmart/rLogin/blob/choose-network/src/controllers/providers.ts#L146). This is how the whole thing works.
* `connectTo` method also accepts the optional parameter. [See here](https://github.com/rsksmart/rLogin/blob/choose-network/src/controllers/providers.ts#L180)

In the future, this file could be used to solve #106, and 
 https://github.com/Web3Modal/web3modal/pull/300

ilanolkies commented on src/Core.tsx:
```diff
@@ -226,15 +239,36 @@ export class Core extends React.Component<IModalProps, IModalState> {
+  private checkIfMultipleNetworks (provider: RLoginIProviderUserOptions) {
+    const { rpcUrls } = this.props
+    // choose the network first:
+    if (['Ledger', 'Trezor', 'Dcent'].includes(provider.name) && rpcUrls) {
```
Should this be configurable?

jessgusclark replied:
I don't think so. This "choose network" feature is only compatible with our hardware providers since it passes the URL and chain Id in a [specific format](https://github.com/rsksmart/rLogin/pull/150/files#diff-f0006202dbed07b20ed558041889df8f0bbce2825b1c0a84bd44728aef0a33a2R187-R190). As such, we can't apply this directly to Portis as they send the chainId and URL in a [network](https://github.com/rsksmart/rLogin/blob/develop/sample/front/index.html#L95-L98) variable.

If the developer does not want to show this modal and only use on the configuration in the initial rLogin setup they can choose not to pass the rpcUrl parameter. Which is the second condition in the if statement above. This is explained in the readme.

ilanolkies reviewed (changes requested):
LGTM!

ilanolkies commented:
Checklist to test the hardware devices:
- [x] Ledger (working with RSK app for RSK Mainnet and Ethereum app for RSK Testnet)
- [ ] Trezor
- [ ] D'Cent

jessgusclark commented:
Rebased to develop and changed the 'preConnect' function name to match #159 naming convention. These two PRs do not depend on each other now and each can be rebased quickly into each other depending on which gets merged first.

ilanolkies reviewed (approved):
🔥 🔥 🔥
